relay-common/src/metrics.rs

```diff
@@ -61,19 +61,59 @@
 //! [`configure_statsd`]: fn.configure_statsd.html
 //! [`metric!`]: ../macro.metric.html
 
+use std::collections::BTreeMap;
 use std::net::ToSocketAddrs;
+use std::ops::{Deref, DerefMut};
 use std::sync::Arc;
 
-use cadence::StatsdClient;
+use cadence::{Metric, MetricBuilder, StatsdClient};
 use lazy_static::lazy_static;
 use parking_lot::RwLock;
 
+/// Client configuration object to store globally.
+#[derive(Clone, Debug)]
+pub struct MetricsClient {
```

Comment:
Can you remove the `Clone` here to make sure that we always only clone the `Arc`?